tests/test_views.py

```diff
@@ -35,4 +37,81 @@ def test_prompt_model_admin_viewset_edit_view(client, setup_users, setup_prompt_
     assert setup_prompt_object.label in str(response.content)
 
 
-# TODO add tests for process view
+@pytest.mark.django_db
+def test_process_view_get_request(client, setup_users):
+    url = reverse("wagtail_ai:process")
+
+    superuser = setup_users
+    client.force_login(superuser)
+
+    response = client.get(url)
+    assert response.status_code == 400
+    assert response.json() == {
+        "error": "No text provided - please enter some text before using AI features"
+    }
```

Comment:
**Suggestion:** We duplicate a lot of code in this file. I think we could use table-driven tests instead: https://docs.pytest.org/en/7.3.x/how-to/parametrize.html.

Reply:
Updated what I could

Reply:
Well, we did not reduce any duplication. Sorry if you've misunderstood my comment. I think it's fine to remove the parametrize functionality and move on so this can get merged.

I had something like this in mind to make the code a bit more DRYer.

```python
import pytest


class ValidPrompt:
    pass


TEST_TABLE = [
    pytest.param(
        ({"text": "text", "prompt": str(uuid.uuid4())}, 400, {"error": "Invalid prompt provided."}),
        id="bad_prompt_id"
    ),
    pytest.param(
        ({"text": "text", "prompt": ValidPrompt}, 200, {"message": "This is an echo backend: test"}),
        id="valid_prompt_id"
    ),
    pytest.param(
        ({}, 400, {"error": "No text provided - please enter some text before using AI features."}),
        id="post_without_text"
    ),
]


@pytest.mark.django_db
@pytest.mark.parametrize("data,expected_status_code,expected_json",TEST_TABLE)
def test_process_view(client, setup_users, setup_prompt_object, data, expected_status_code, expected_json):
    url = reverse("wagtail_ai:process")

    superuser = setup_users
    client.force_login(superuser)

    if data['prompt'] is ValidPrompt :
        data["prompt"] = str(setup_prompt_object.uuid)

    response = client.post(url, data=data)
    assert response.status_code == expected_status_code
    if expected_json is not None:
        assert response.json() == expected_json
```

Ref: https://blog.jetbrains.com/go/2022/11/22/comprehensive-guide-to-testing-in-go/#writing-table-driven-tests

Reply:
Thanks!